src/resolver.cpp

```diff
@@ -235,211 +222,95 @@ std::string UsdOpenAssetIOResolver::_CreateIdentifier(
-  logger_->debug("  result: " + result.GetPathString());
-
-  return result;
+  if (manager_->isEntityReferenceString(assetPath)) {
+    std::string message = "Writes to OpenAssetIO entity references are not currently supported ";
+    message += assetPath;
+    logger_->critical(message);
+    return ArResolvedPath("");
+  }
+  return ArDefaultResolver::_ResolveForNewAsset(assetPath);
 }
 
-/* Asset Operations*/
+/*
+ * Pass-through asset operations
+ *
+ * These methods are simply relayed to the ArDefaultResolver. There may
+ * be interest in fetching data for some of these from the manager, but
+ * we don't have a real use case just yet. Doing so increases complexity
+ * as we'd need to return both the resolved path _and_ the original
+ * entity reference from _Resolve, so we could make queries in these
+ * methods. We'll need this for publishing operations, but avoiding that
+ * overhead for the more common (and hot) read case is critical.
+ *
+ */
 std::string UsdOpenAssetIOResolver::_GetExtension(const std::string &assetPath) const {
-  const std::string fnName = TF_FUNC_NAME();
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] " << fnName
-           << "\n  assetPath: " << assetPath;
-    logger_->debug(logMsg.str());
-  }
-
-  auto result = catchAndLogExceptions(
-      [&] {
-        // TODO(DF): Give the manager a chance to provide the file extension.
-        return ArDefaultResolver::_GetExtension(
-            locationInManagerContextForEntity(manager_, readContext_, assetPath));
-      },
-      logger_, fnName);
-
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] "
-           << "result: " << result;
-    logger_->debug(logMsg.str());
-  }
-
-  return result;
+  return ArDefaultResolver::_GetExtension(assetPath);
 }
 
 ArAssetInfo UsdOpenAssetIOResolver::_GetAssetInfo(const std::string &assetPath,
                                                   const ArResolvedPath &resolvedPath) const {
-  const std::string fnName = TF_FUNC_NAME();
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] " << fnName
-           << "\n  assetPath: " << assetPath
-           << "\n  resolvedPath: " << resolvedPath.GetPathString();
-    logger_->debug(logMsg.str());
-  }
-  // TODO(DF): Determine if there is value in supporting this via the
-  //  manager, including any useful data that can be stuffed into the
-  //  generic `resolverInfo` VtValue member.
-  auto result = ArDefaultResolver::_GetAssetInfo(assetPath, resolvedPath);
-
-  {
-    std::ostringstream logMsg;
-    logMsg << "[tid=" << std::this_thread::get_id() << "] "
-           << "\n  result.assetName: " << result.assetName
-           << "\n  result.repoPath: " << result.repoPath;
-    logger_->debug(logMsg.str());
-  }
-
-  return result;
+  return ArDefaultResolver::_GetAssetInfo(assetPath, resolvedPath);
```

Comment:
[question] Are the arguments to this as we would expect, i.e. so that we still have access to the entity reference despite the changes to  `_Resolve`, in case we ever want return a useful response here?

Reply:
Just saw the docstring in the final commit, which justifies why we don't really care right now if this could work with the manager or not.

Reply:
Yeah - I'm not 100% on the use cases behind this function. Was kind of hoping we could keep its as a passthrough until we know of a workflow where we specifically need to get this info from the manager. I'm hoping that `assetPath` would be the input, and `resolvedPath` would be what comes out of `_Resolve`.